sky/spot/spot_state.py

```diff
@@ -308,79 +377,119 @@ def set_cancelled(job_id: int):
         """\
         UPDATE spot SET
         status=(?), end_at=(?)
-        WHERE job_id=(?) AND status=(?)""",
+        WHERE new_job_id=(?) AND status=(?)""",
         (SpotStatus.CANCELLED.value, time.time(), job_id,
          SpotStatus.CANCELLING.value))
     _CONN.commit()
     logger.info('Job cancelled.')
 
 
 # ======== utility functions ========
-
-
 def get_nonterminal_job_ids_by_name(name: Optional[str]) -> List[int]:
     """Get non-terminal job ids by name."""
-    name_filter = 'AND job_name=(?)' if name is not None else ''
     field_values = [status.value for status in SpotStatus.terminal_statuses()]
+
+    name_filter = ''
     if name is not None:
-        field_values.append(name)
+        name_filter = 'AND (spot.job_name=(?) OR job_names.name=(?))'
```

Comment:
I think it should be cleaner to have two tables to avoid storing duplicated information in a single table. Also, make it more extensible when we have more job info.